src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -42,35 +43,25 @@ template Rule
     -- Lifecycle time update events
     interface instance Lifecycle.I for Rule where
       view = Lifecycle.View with lifecycler; id; description
-      -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_BEGIN
+      -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_BEGIN
       evolve Lifecycle.Evolve{eventCid; observableCids; instrument} = do
-        claimInstrument <- fetchInterfaceByKey @BaseInstrument.R instrument
-      -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_END
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_FASTFORWARD_BEGIN
-        -- Recover claims tree as of the lastEventTimestamp. For a bond, this just requires
-        -- lifecycling as of the lastEventTimestamp.
-        nv <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I instrument lifecycler
-          DynamicInstrument.GetView with viewer = lifecycler
-        claims <- fst <$>
-          lifecycle lifecycler observableCids claimInstrument [timeEvent nv.lastEventTimestamp]
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_FASTFORWARD_END
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_BEGIN
-        -- Lifecycle
-        let acquisitionTime = Claim.getAcquisitionTime claimInstrument
         v <- view <$> fetch eventCid
-        (remaining, pending) <- lifecycleClaims observableCids acquisitionTime claims
-          [timeEvent v.eventTime]
+        (remaining, pending, claims, claimInstrument, nv) <-
+          fastForwardAndLifecycle instrument observableCids v.eventTime lifecycler
+
         let
           pendingAfterNetting = netOnTag pending
           (otherConsumed, otherProduced) = splitPending pendingAfterNetting
         if remaining == claims && null pendingAfterNetting then
           pure (None, [])
         else do
           let
+            electionData = None
             currentKey = BaseInstrument.getKey $ toInterface claimInstrument
             newKey = currentKey with version = sha256 $ show remaining
             producedInstrument = if isZero' remaining then None else Some newKey
-          tryCreateNewInstrument lifecycler v.eventTime instrument newKey
+          tryCreateNewInstrument lifecycler nv.prevElections v.eventTime electionData instrument
```

Comment:
```suggestion
          tryCreateNewInstrument lifecycler nv.prevElections v.eventTime None instrument
```

and remove `electionData = None`

Reply:
Also, in case of an election event, aren't we creating an unnecessary instrument here?

We create 
- one instrument now
- one after the election is processed

We should try to create only one instrument at the end